### DISCUSSION TITLE
How to apply filter in loop for all fields?

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#1014] [Q&A] [answered]
Hi,
I have this CSV input

```
a,b,c
,2,3
1,,
1,2,3
```

I would like - in example - to run a miller filter to have in output all the records in which at least one field is empty. I could run a bash for loop, but I'm sure there is a Miller way to do it.

And it's not

```
 mlr -S --csv filter  '
  for (k in $*) {
    $[k] =~ "^$";
  }
' input.csv
```

Thank you

**Comment by johnkerl** [comment, marked as answer]
Hi @aborruso ! I would suggest either of:

```
$ mlr --csv --from input.csv filter 'any($*, func(k,v) {return v==""})'
a,b,c
,2,3
1,,
```

```
$ mlr --csv --from input.csv filter 'any($*, func(k,v) {return is_empty(v)})'
a,b,c
,2,3
1,,
```

**Reply by aborruso** [reply]
Dear @johnkerl I think you should add it as filter example in the documentation.
It could be very useful

**Comment by aborruso** [comment]
It's a kind of magic!!!

Thank you
